### PR TITLE
NEW: Navigation components

### PR DESCRIPTION
**Pull request by vepor**
New components for navigation purposes:
- NavigationBar
- NavigationList
- NavigationListItem
- Drawer

This Pull Request meets the following criteria:

* [x] Tests have been added/adjusted for my new feature
* [x] New Components are registered in index.js of my project
<br/><br/><br/><url>LiveURL: https://orbit-components-add-navigation-components.surge.sh</url>

### CODE REVIEW

**Comment by mainframev** [comment]
jeeez. that looks awesome

File: src/DestinationHeader/index.js [hunk 1]
```diff
@@ -97,7 +97,7 @@ StyledHeader.defaultProps = {
 };
 
 const DestinationHeaderGoBackButton = ({ onClick }) => {
-  const dictionary = React.useContext(DictionaryContext);
+  const dictionary = useDictionary();
```
Comment (filipdanisko) [inline src/DestinationHeader/index.js]: 💯

File: src/NavigationList/NavigationListItem/index.js [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+StyledNavigationListItemIcon.defaultProps = {
+  theme: defaultTheme,
+};
+
+const StyledNavigationListItemContent = styled.div`
+  font-family: ${({ theme }) => theme.orbit.fontFamily};
+  font-size: ${({ theme }) => theme.orbit.fontSizeTextNormal};
+  font-weight: ${({ theme }) => theme.orbit.fontWeightMedium};
+  line-height: 20px;
```
[the remark this answers is not in the document]
Comment (vepor) [reply src/NavigationList/NavigationListItem/index.js]: Not released yet.

File: src/NavigationBar/README.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
+# NavigationBar
+To implement NavigationBar component into your project you'll need to add the import:
+```jsx
+import NavigationBar from "@kiwicom/orbit-components/lib/NavigationBar";
+```
+After adding import into your project you can use it simply like:
+```jsx
+<NavigationBar onMenuOpen={handleDrawerOpen}>
+  <NavigationLink>About us</NavigationLink>
```
[the remark this answers is not in the document]
Comment (vepor) [reply src/NavigationBar/README.md]: 🙈 I forgot to change docs, there's no `NavigationLink` anymore.

File: src/Drawer/components/DrawerClose.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+import styled from "styled-components";
+
+import ButtonLink from "../../ButtonLink/index";
+import ChevronRight from "../../icons/ChevronRight";
+import Close from "../../icons/Close";
+import type { Props } from "./DrawerClose";
+import { isNavigation } from "../helpers/isType";
+import defaultTheme from "../../defaultTheme";
+import Translate, { pureTranslate } from "../../Translate";
+import { DictionaryContext } from "../../Dictionary/index";
```
[the remark this answers is not in the document]
Comment (vepor) [reply src/Drawer/components/DrawerClose.js]: Probably, yes. But it really doesn't matter even if `eslint` likes it 🙂

**Pull request merged**